Engine/Storage/LocalObjectStore.cs

```diff
@@ -0,0 +1,335 @@
+            }
+
+            return _storage.ContainsKey(key);
+        }
+
+        /// <summary>
+        /// Returns the object data for the specified key
+        /// </summary>
+        /// <param name="key">The object key</param>
+        /// <returns>A byte array containing the data</returns>
+        public byte[] ReadBytes(string key)
+        {
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            byte[] data;
+            if (!_storage.TryGetValue(key, out data))
+            {
+                throw new KeyNotFoundException($"Object with key '{key}' was not found in the current project. " +
+                    "Please use ObjectStore.ContainsKey(key) to check if an object exists before attempting to read."
+                );
+            }
+
+            return data;
+        }
+
+        /// <summary>
+        /// Saves the object data for the specified key
+        /// </summary>
+        /// <param name="key">The object key</param>
+        /// <param name="contents">The object data</param>
+        /// <returns>True if the save operation was successful</returns>
+        public bool SaveBytes(string key, byte[] contents)
+        {
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            var fileCount = 0;
+            var expectedStorageSizeBytes = 0L;
+            foreach (var kvp in _storage)
+            {
+                fileCount++;
+                if (string.Equals(kvp.Key, key))
+                {
+                    expectedStorageSizeBytes += contents.Length - kvp.Value.Length;
+                }
+                else
+                {
+                    expectedStorageSizeBytes += kvp.Value.Length;
+                }
+            }
+
+            if (fileCount > Controls.StorageFileCount)
+            {
+                Log.Error($"LocaObjectStore at file capacity: {fileCount}. Unable to save: '{key}'");
+                return false;
+            }
+
+            var expectedStorageSizeMb = BytesToMb(expectedStorageSizeBytes);
+            if (expectedStorageSizeMb > Controls.StorageLimitMB)
+            {
+                Log.Error($"LocalObjectStore at storage capacity: {expectedStorageSizeMb}. Unable to save: '{key}'");
+                return false;
+            }
+
+            _dirty = true;
+            _storage.AddOrUpdate(key, k => contents, (k, v) => contents);
+            return true;
+        }
+
+        /// <summary>
+        /// Deletes the object data for the specified key
+        /// </summary>
+        /// <param name="key">The object key</param>
+        /// <returns>True if the delete operation was successful</returns>
+        public bool Delete(string key)
+        {
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
```

Comment:
I believe the dictionary will already throw `ArgumentNullException` if the key is null, not sure we need these extra checks? idem for the other methods